src/FileLoaderHelper.h

```diff
@@ -151,4 +157,12 @@ inline void EnsureFileFromURLExists(std::string url, std::string file, std::stri
     printout(ERROR, "FileLoader", "check permissions and retry");
     std::_Exit(EXIT_FAILURE);
   }
+
+  // final check of the file size
+  if (fs::file_size(file_path) == 0) {
```

Comment:
I see, in https://en.cppreference.com/w/cpp/filesystem/file_size they say

> The result of attempting to determine the size of a directory (as well as any other file that is not a regular file or a symlink) is implementation-defined.

I'm not super familiar with std::filesystem, but perhaps it would be tiny bit safer to have:
```suggestion
  if (fs::file_size(fs::canonical(file_path)) == 0) {
```

Reply:
Oh, misread it. I guess it says that symlinks are fine.

Reply:
> symlinks are followed

Yeah